packages/page-migration/src/Migration/index.tsx

```diff
@@ -0,0 +1,261 @@
+// Copyright 2017-2022 @polkadot/app-rpc authors & contributors
+// SPDX-License-Identifier: Apache-2.0
+
+import React, { useState, useEffect } from 'react';
+
+import type { DeriveBalancesAll } from '@polkadot/api-derive/types';
+
+import { Available, Dropdown, InputAddressCheqd } from '@polkadot/react-components';
+import { InputAddress, MarkError, MarkWarning, TxButton, Button } from '@polkadot/react-components';
+
+import { useTranslation } from '../translate';
+import { isValidCheqdAddress, getCheqdAddressError } from '@polkadot/react-components/InputAddressCheqd';
+import { useApi, useCall } from '@polkadot/react-hooks';
+import { formatBalance } from '@polkadot/util';
+import MarkSuccess from '@polkadot/react-components/MarkSuccess';
+
+function convertDockToCheqd(dockBalance) {
+  const swapRatio = 18.5178;
+  const dockAmt = dockBalance.toNumber() / 1000000;
+  const cheqdBalance = dockAmt / swapRatio;
+  return cheqdBalance;
+}
+
+function MigrationApp ({ className }): React.ReactElement {
+  const { t } = useTranslation();
+  const [senderId, setSenderId] = useState<string | null>(null);
+  const [cheqdId, setCheqdId] = useState<string>('');
+  const [success, setSuccess] = useState<boolean>(false);
+  const [selectFromDropdown, setSelectFromDropdown] = useState<boolean>(false);
+  const [cheqdAddresses, setCheqdAddresses] = useState<string[]>([]);
+  const { api } = useApi();
+  const allBalances = useCall<DeriveBalancesAll>(api.derive.balances?.all, [senderId]);
+
+  const values = [cheqdId];
+  const extrinsic = api && api.tx && api.tx.cheqdMigration && api.tx.cheqdMigration.migrate(values);
+  const isValid = senderId && cheqdId && isValidCheqdAddress(cheqdId);
+
+  function handleSuccess() {
+    setSuccess(true);
+  }
+
+  function handleRestart() {
+    setSuccess(false);
+    setCheqdId('');
+    setSenderId(null);
+  }
+
+  async function getExtensionAddresses() {
+    const chainId = 'cheqd-mainnet-1';
+    const walletObj = window.keplr || window.leap;
+
+    await walletObj.enable(chainId);
+
+    const offlineSigner = walletObj.getOfflineSigner(chainId);
+    const accounts = await offlineSigner.getAccounts();
+    setCheqdAddresses([...cheqdAddresses, ...accounts.map(({ address }) => address)]);
+    setSelectFromDropdown(accounts.length > 0);
+  }
+
+  useEffect(() => {
+    if (window.keplr) {
```

Comment:
Shouldn't this be `window.keplr || window.leap`?

Reply:
yeah i guess, so the interesting thing is both those wallets pretend to be eachother and inject the same variables (keplr and leap) so it does work as is. ill update for posterity though